server/bundles/io.cloudbeaver.service.security/src/io/cloudbeaver/service/security/CBEmbeddedSecurityController.java

```diff
@@ -1407,21 +1407,41 @@ private SMAuthInfo finishAuthentication(
         Map<SMAuthConfigurationReference, Object> storedUserData = new LinkedHashMap<>();
         SMTeam[] allTeams = null;
         String detectedAuthRole = null;
+        Map<String, Object> providerConfig = new LinkedHashMap<>();
+        Map<String, Object> userAuthData = new LinkedHashMap<>();
         for (SMAuthConfigurationReference authConfiguration : authProviderIds) {
             String authProviderId = authConfiguration.getAuthProviderId();
             WebAuthProviderDescriptor authProvider = getAuthProvider(authProviderId);
-            var userAuthData = (Map<String, Object>) authInfo.getAuthData().get(authConfiguration);
-            SMAutoAssign autoAssign = getAutoAssignUserData(authId, authProvider, userAuthData, finishAuthMonitor);
+
+            {
+                // Read auth provider configuration.
+                // Note: if there are several auth providers in a row then we'll reuse provider config from previous one.
+                // This is how federated auth + native auth providers work as a couple.
+                String providerConfigId = readProviderConfigId(authInfo.getAuthAttemptId(), authProvider.getId());
```

Comment:
extra reading from database, we already have provider config id in SMAuthConfigurationReference